lib/validation/validator.js

```diff
@@ -0,0 +1,61 @@
+const Ajv = require("ajv");
+const ajvErrors = require("ajv-errors");
+const path = require("path");
+const {promisify} = require("util");
+const readFile = promisify(require("fs").readFile);
+
+const ValidationError = require("./ValidationError");
+
+async function loadSchema(schemaPath) {
+	const filePath = schemaPath.replace("http://ui5.sap/schema/", "");
+	const schemaFile = await readFile(path.join(__dirname, "schema", filePath), {encoding: "utf8"});
+	return JSON.parse(schemaFile);
+}
+
+class Validator {
```

Comment:
Same here, no  documentation. Most of the methods logic is obvious but at least the class could use some doc.